xbmc/Application.cpp

```diff
@@ -3232,11 +3225,27 @@ PlayBackRet CApplication::PlayFile(CFileItem item, const std::string& player, bo
 
   // We should restart the player, unless the previous and next tracks are using
   // one of the players that allows gapless playback (paplayer, VideoPlayer)
-  // DVD playback does not support gapless
-  if (item.IsDiscImage() || item.IsDVDFile())
-    m_appPlayer.ClosePlayer();
-  else
-    m_appPlayer.ClosePlayerGapless(newPlayer);
+  if (m_appPlayer.HasPlayer())
+  {
+    if (m_appPlayer.CanClosePlayerGapless(newPlayer, item))
```

Comment:
looking at the names of the methods, the logic is not clear to me. when can a gapless player be closed and why do we want to close it?

Reply:
gapless close is just close file.

Reply:
then a more appropriate name would be CloseFile or something, wouldn't it?

Reply:
This is just the test, "can you close gapless" so perhaps "SupportsGaplessClose" or something?

Reply:
the method itself returns "gaplessSupported" and this seems what the methods actually checks. the words "can" and "close" are not related to what the method does.

Reply:
I've renamed it (pushed 2nd commit) to "SupportsGaplessClose". IMO this is a better name, because it returns true is the old player, the new player in combination with the new item supports a gapless transition (i.e. close of the old file).....